Pod/Classes/SocketManager.swift

```diff
@@ -92,16 +98,21 @@ public  class SocketManager {
                         self.experienceChannel?.onRequest(response)
                         self.organizationChannel?.onRequest(response)
                     }else{
-                        let channelEnum:SOCKET_CHANNELS = SOCKET_CHANNELS(rawValue: channel as! String)!
-                        switch channelEnum{
-                            case .SYSTEM :
-                                self.systemChannel?.onRequest(response)
-                            case .LOCATION:
-                                self.locationChannel?.onRequest(response)
-                            case .EXPERIENCE:
-                                self.experienceChannel?.onRequest(response)
-                            case .ORGANIZATION:
-                                self.organizationChannel?.onRequest(response)
+                        if(self.channelCache.indexForKey(channel as! String) != nil){
```

Comment:
I think it doesn't matter because to get the socket channels that are available in the SDK you need to pass an ENUM and for getting any channel you do it with STRING.